WordPress/openverse-frontend
# Give licenses the full name and retire `LicenseIcons`

## Fixes


Fixes #358 by @obulat

## Description


This PR helps describe better the license in the `VLicense` component adding an `aria-label` with the readable name of the license. This way using VoiceOver in the filter sidebar you should hear the full name of the license instead of the spelling of the letters one by one.

It also replaces the old `LicenseIcons` component in favor of this one. For this, two properties had to be added: `hideName` and `filledBg`, more details in the Storybook story.

## Testing Instructions

Run storybook and see the new VLicense story showing only icons.
```
npm run storybook
```
Also, run the project normally and verify the replaced component in the Search image results page and in the individual result page.
```
npm run dev
```

<img width="1417" alt="CleanShot 2021-11-17 at 18 34 50@2x" src="https://user-images.githubusercontent.com/9145885/142293509-fd54a992-06e9-441d-b376-a1cbf40beadf.png">

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
